### PR TITLE
Add transform data functionality for scatter plot

### PR DESCRIPTION
**Pull request by kat-star**
[Scatter Plot] Added transform data functionality for aggregate and attribute queries

https://newrelic.atlassian.net/browse/DEVEX-1818

### CODE REVIEW

File: visualizations/scatter-plot-chart/index.js [hunk 1]
```diff
@@ -32,6 +33,69 @@ export default class ScatterPlotChartVisualization extends React.Component {
...
+      ([facetGroupName, facetGroupData]) => ({
+        facetGroupName,
+        ...facetGroupData,
+      })
+    );
+  };
+
+  getNonAggregatesData = (rawData) => {
+    const { uniqueNonAggregates } = getUniqueNonAggregates(rawData);
+    const attributeNames = Array.from(uniqueNonAggregates);
```
Comment (timglaser) [inline visualizations/scatter-plot-chart/index.js:70]: Like we discussed, we're better off not deduplicating the aggregates.

**Review by timglaser — CHANGES_REQUESTED** [review]
Thanks for taking this on! A few thoughts here to consider.

File: visualizations/scatter-plot-chart/index.js [hunk 2]
```diff
@@ -86,22 +155,19 @@ export default class ScatterPlotChartVisualization extends React.Component {
...
-                      { x: 3, y: 5 },
-                      { x: 4, y: 4 },
-                      { x: 5, y: 7 },
-                    ]}
+                    data={series}
+                    style={{
+                      data: {
+                        fill: ({ datum }) => datum.color,
+                        fillOpacity: 0.7,
+                        strokeWidth: 3,
```
[the remark this answers is not in the document]
Comment (kkaiyang94) [reply visualizations/scatter-plot-chart/index.js]: Sounds good!

File: visualizations/scatter-plot-chart/index.js [hunk 3]
```diff
@@ -32,6 +33,74 @@ export default class ScatterPlotChartVisualization extends React.Component {
...
+        ...facetGroupData,
+      })
+    );
+  };
+
+  getNonAggregatesData = (rawData) => {
+    const { uniqueNonAggregates } = getUniqueNonAggregates(rawData);
+    const nonAggregates = {};
+
+    Array.from(uniqueNonAggregates).map((datapoint, i) => {
```
[the remark this answers is not in the document]
Comment (kkaiyang94) [reply visualizations/scatter-plot-chart/index.js]: Refactored this part to reflect the comments

File: visualizations/scatter-plot-chart/index.js (hunk 1 above)
Comment (kkaiyang94) [reply visualizations/scatter-plot-chart/index.js:70]: It appears that the NrqlQuery component is deduplicating the aggregates.

**Review by timglaser — APPROVED** [review]
Looks good!

**Pull request merged**